File: Frontend-POS/RBMS-POS-Client/src/app/core/api/models/option-group-response-model.ts
/* tslint:disable */
/* eslint-disable */
import { LinkedMenuModel } from '../models/linked-menu-model';
import { OptionItemResponseModel } from '../models/option-item-response-model';
export interface OptionGroupResponseModel {
  categoryType?: number;
  categoryTypeName?: string | null;
  createdAt?: string;
  createdBy?: string | null;
  isActive?: boolean;
  isRequired?: boolean;
  linkedMenuCount?: number;
  linkedMenus?: Array<LinkedMenuModel> | null;
  maxSelect?: number | null;
  minSelect?: number;
  name?: string | null;
  optionGroupId?: number;
  optionItems?: Array<OptionItemResponseModel> | null;
  sortOrder?: number;
  updatedAt?: string | null;
  updatedBy?: string | null;
}

File: Backend-POS/POS.Main/POS.Main.Business.Menu/Models/OptionGroup/OptionGroupResponseModel.cs
namespace POS.Main.Business.Menu.Models.OptionGroup;

public class OptionGroupResponseModel
{
    public int OptionGroupId { get; set; }
    public string Name { get; set; } = string.Empty;
    public int CategoryType { get; set; }
    public string CategoryTypeName { get; set; } = string.Empty;
    public bool IsRequired { get; set; }
    public int MinSelect { get; set; }
    public int? MaxSelect { get; set; }
    public int SortOrder { get; set; }
    public bool IsActive { get; set; }
    public List<OptionItemResponseModel> OptionItems { get; set; } = new();
    public int LinkedMenuCount { get; set; }
    public List<LinkedMenuModel>? LinkedMenus { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }
    public string? CreatedBy { get; set; }
    public string? UpdatedBy { get; set; }
}

public class OptionItemResponseModel
{
    public int OptionItemId { get; set; }
    public string Name { get; set; } = string.Empty;
    public decimal AdditionalPrice { get; set; }
    public decimal? CostPrice { get; set; }
    public int SortOrder { get; set; }
    public bool IsActive { get; set; }
}

public class LinkedMenuModel
{
    public int MenuId { get; set; }
    public int? ImageFileId { get; set; }
    public string NameThai { get; set; } = string.Empty;
    public string NameEnglish { get; set; } = string.Empty;
    public string CategoryTypeName { get; set; } = string.Empty;
    public bool IsAvailable { get; set; }
}
